File: mobile/lib/features/b2b/domain/entities/order_status.dart
enum OrderStatus {
  draft,
  pending,
  confirmed,
  processing,
  ready,
  shipped,
  delivered,
  cancelled,
  refunded;

  String get displayName {
    switch (this) {
      case OrderStatus.draft:
        return 'Brouillon';
      case OrderStatus.pending:
        return 'En attente';
      case OrderStatus.confirmed:
        return 'Confirmée';
      case OrderStatus.processing:
        return 'En préparation';
      case OrderStatus.ready:
        return 'Prête';
      case OrderStatus.shipped:
        return 'Expédiée';
      case OrderStatus.delivered:
        return 'Livrée';
      case OrderStatus.cancelled:
        return 'Annulée';
      case OrderStatus.refunded:
        return 'Remboursée';
    }
  }

  bool get canCancel {
    return this == OrderStatus.draft ||
        this == OrderStatus.pending ||
        this == OrderStatus.confirmed;
  }

  bool get canModify {
    return this == OrderStatus.draft || this == OrderStatus.pending;
  }

  bool get isTerminal {
    return this == OrderStatus.delivered ||
        this == OrderStatus.cancelled ||
        this == OrderStatus.refunded;
  }

  static OrderStatus fromString(String value) {
    return OrderStatus.values.firstWhere(
      (e) => e.name.toUpperCase() == value.toUpperCase(),
      orElse: () => OrderStatus.pending,
    );
  }
}

File: backend/jufa-backend/src/main/java/ml/jufa/backend/b2b/entity/OrderStatus.java
package ml.jufa.backend.b2b.entity;

public enum OrderStatus {
    DRAFT("Brouillon"),
    PENDING("En attente"),
    CONFIRMED("Confirmée"),
    PROCESSING("En préparation"),
    READY("Prête"),
    SHIPPED("Expédiée"),
    DELIVERED("Livrée"),
    CANCELLED("Annulée"),
    REFUNDED("Remboursée");

    private final String displayName;

    OrderStatus(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }

    public boolean canCancel() {
        return this == DRAFT || this == PENDING || this == CONFIRMED;
    }

    public boolean canModify() {
        return this == DRAFT || this == PENDING;
    }

    public boolean isTerminal() {
        return this == DELIVERED || this == CANCELLED || this == REFUNDED;
    }
}
